packages/app/src/cli/services/bulk-operations/watch-bulk-operation.ts

```diff
@@ -0,0 +1,69 @@
+import {formatBulkOperationStatus} from './format-bulk-operation-status.js'
+import {
+  GetBulkOperationById,
+  GetBulkOperationByIdQuery,
+} from '../../api/graphql/bulk-operations/generated/get-bulk-operation-by-id.js'
+import {adminRequestDoc} from '@shopify/cli-kit/node/api/admin'
+import {sleep} from '@shopify/cli-kit/node/system'
+import {AdminSession} from '@shopify/cli-kit/node/session'
+import {outputContent} from '@shopify/cli-kit/node/output'
+import {renderSingleTask} from '@shopify/cli-kit/node/ui'
+
+const TERMINAL_STATUSES = ['COMPLETED', 'FAILED', 'CANCELED', 'EXPIRED']
+const POLL_INTERVAL_SECONDS = 5
```

Comment:
I guess many operations will be faster than 5 seconds. Couldn't we poll more frequently?

Or even better: what about a dynamic interval? I would poll every 1-2 seconds during the first 10-20 seconds, so we show enough feedback for fast queries, and then every 5s.

Otherwise, for queries under 5s, we will just show `0 objects` and then `Completed`.

Reply:
That's a great idea, ty!!! Noted as a polish improvement [here](https://github.com/shop/issues-api-foundations/issues/1149), will add as a follow-up.